### DISCUSSION TITLE
New Release?

### DISCUSSION THREAD

**Discussion by Alexhuszagh** [cross-rs/cross#761] [Ideas]
Due to the number of bugs fixed, and the lack of updates on the images but move to a new container registry, I feel it's probably a good idea to draft a new release by early July (v0.2.2), and then we can deal with backwards-incompatible changes for v0.3.0? This way the bug fixes can be provided for the default install on crates.io, so users won't have to install from git. A lot of bugs have been patched, so I think this is probably a sound roadmap.

Thoughts?

A list of goals for the next release are:

I think the major goals remaining for 0.2.2 are getting:
- ~~`x86_64-unknown-linux-gnu` based on Ubuntu, and keeping at least it or another image compatible with RHEL7 (glibc 2.17) (#764).~~
- ~~Add `build-std` support so we can rely less (hopefully, never) on `xargo` (#782).~~
- ~~Probably add `Cargo.toml` as a configuration source, if possible (#754).~~

More stretch goals but we're getting closer to are:
- ~~Remote docker support (I almost have full support ready, #785).~~
- ~~Custom dockerfile/pre-build hooks without having to maintain images (#678).~~

Also, we need to ensure we instruct users to install from crates.io prior to release (IE, not from git).

Update: we just pushed a [release candidate](https://github.com/cross-rs/cross/pull/855).

**Comment by Emilgardis** [comment]
#542 ping

yes, I think it's time, I'm still very hesitant about the x86_64-unknown-linux-gnu changes though (#669 and #680 ) .

**Reply by Alexhuszagh** [reply]
We could possibly skip CentOS altogether just by both having GCC and glibc built from source for just that target, which is the single best-supported target arch/OS combination, and we won't have top do a multi-pass build since the host triple is the same as the target triple.

**Comment by otavio** [comment]
Could you guys review if there are any still need fix for 0.2.2 and put it on the specific milestone? and the same for 0.3.0?

**Reply by Alexhuszagh** [reply]
I think the major goals remaining for 0.2.2 are getting:
- `x86_64-unknown-linux-gnu` based on Ubuntu, and keeping at least it or another image compatible with RHEL7 (glibc 2.17).
- ~~Add `build-std` support so we can rely less (hopefully, never) on `xargo`.~~
- Probably add `Cargo.toml` as a configuration source, if possible.

More stretch goals but we're getting closer to are:
- Remote docker support (I almost have full support ready).
- ~~#678 for custom dockerfile/pre-build hooks without having to maintain images.~~

We've got some issues with the Apple M1 processors, but I'm not sure if anyone else besides Emil can debug those right now.

**Reply by Alexhuszagh** [reply]
Updates: #754 is almost ready to merge, #764 is about to be merged, and #785 seems to be getting much more robust. That means most of our immediate issues preventing a new release are almost complete.

**Comment by naiba** [comment]
Can't wait. The `pre-build` is very useful, I was experimenting with the contents of the main branch README.md many times and finally found that it was not yet released 😨.

**Comment by naiba** [comment]
https://github.com/cross-rs/cross/milestone/1

Looks like nothing is stopping v0.2.2 from being released?

**Comment by naiba** [comment]
Hello, what's the next plan? I'm very eager to use v0.2.2, as actions-rs/cargo can't customize the cross's version. I think the ease of cross-compiling in GitHub Actions will be very helpful in building the rust community. ❤️

**Reply by Alexhuszagh** [reply]
> Hello, what's the next plan? I'm very eager to use v0.2.2, as actions-rs/cargo can't customize the cross's version. I think the ease of cross-compiling in GitHub Actions will be very helpful in building the rust community. heart

We're patching a few minor issues, scheduling a beta/staging release to ensure our infrastructure is correct, and then going to do the release. All the major obstacles have been fixed, so this occur soon.

**Reply by Emilgardis** [reply]
A correction also, you can customize the version as I've detailed in https://github.com/cross-rs/cross/issues/542#issuecomment-1109077784

just download cross before your `cross: true` cargo action

**Comment by Alexhuszagh** [comment]
[v0.2.2](https://crates.io/crates/cross/0.2.2) has been released.

**Reply by naiba** [reply]
awesome
